core/templates/dev/head/pages/topic_editor/TopicEditorStateService.js

```diff
@@ -321,6 +323,21 @@ oppia.factory('TopicEditorStateService', [
           commitMessage, UndoRedoService.getCommittableChangeList()).then(
           function(topicBackendObject) {
             _updateTopic(topicBackendObject);
+            var changeList = UndoRedoService.getCommittableChangeList();
+            for (var i = 0; i < changeList.length; i++) {
+              if (changeList[i].property_name === 'canonical_story_ids') {
+                if (changeList[i].new_value.length <
+                    changeList[i].old_value.length) {
+                  deletedStoryId = changeList[i].old_value.filter(
+                    function(storyId) {
+                      return changeList[i].new_value.indexOf(storyId) === -1;
+                    }
+                  )[0];
```

Comment:
This gets a bit iffy if the user then discards changes to the topic, right?

Might be another reason to treat deletion of stories as its own changelist.

Reply:
This is done only after the user clicks on 'Save Draft'. So, the story is not deleted in the backend until the user has finalised all the changes. 
Also, what did you mean by 'Discard Changes'? I don't think that is present for the topic editor yet.

Reply:
If you have a "Save Draft", you'll need to have a "Discard Changes", I think (that discards the pending draft changes and restores the editor to a clean state). Otherwise, what should the creator do if they decide they don't want their draft after all? Without a "discard changes" the only option is to commit the existing changes, despite them possibly being wrong/incorrect.

The alternative is to just save all changes atomically, i.e. every action done by the creator is immediately saved.

Which of these paradigms are we going with for this page?

Reply:
Yeah, a discard changes seems fine. Will add that,
In that case also, if the changes are discarded, the backend won't be affected, since this request is done only after the user saves the changes.

Reply:
Done. added discard changes.